src/network_service/sending_targets_cache.rs

```diff
@@ -106,28 +100,58 @@ impl SendingTargetsCache {
     }
 }
 
-fn set_target_failed(targets: &mut Vec<(ConnectionInfo, TargetState)>, target: &SocketAddr) {
+/// Info about failed target.
+pub struct TargetFailed {
+    /// Next target to try to resend the message to, if any.
+    pub next: Option<ConnectionInfo>,
+    /// Whether the failed target should be considered as lost - that is - whether it failed more
+    /// than maximum allowed number of times.
```

Comment:
It seem that unless the message is for all targets, this will not return false unless multiple nodes do not respond.
Not clear what is the design here? do we get any information out of messages for 3 out of 7 targets?

Reply:
> do we get any information out of messages for 3 out of 7 targets?

Probably not. Should work for direct messages though.

Reply:
Yes, it will work for direct message, seem a bit strange.

Reply:
Yes. This might not be needed after all. See this: https://github.com/maidsafe/routing/pull/2038#discussion_r377130248